tls/tls-full/config_template.yaml

```diff
@@ -0,0 +1,261 @@
+                maxIdleConns: {{ default .Env.SQL_VIS_MAX_IDLE_CONNS "20" }}
+                maxConnLifetime: {{ default .Env.SQL_VIS_MAX_CONN_TIME "1h" }}
+        {{- else if eq $db "postgresql" }}
+        default:
+            sql:
+                pluginName: "postgres"
+                databaseName: {{ default .Env.DBNAME "temporal" }}
+                connectAddr: "{{ default .Env.POSTGRES_SEEDS "" }}:{{ default .Env.DB_PORT "5432" }}"
+                connectProtocol: "tcp"
+                user: {{ default .Env.POSTGRES_USER "" }}
+                password: {{ default .Env.POSTGRES_PWD "" }}
+                maxConns: {{ default .Env.SQL_MAX_CONNS "20" }}
+                maxIdleConns: {{ default .Env.SQL_MAX_IDLE_CONNS "20" }}
+                maxConnLifetime: {{ default .Env.SQL_MAX_CONN_TIME "1h" }}
+        visibility:
+            sql:
+                pluginName: "postgres"
+                databaseName: {{ default .Env.VISIBILITY_DBNAME "temporal_visibility" }}
+                connectAddr: "{{ default .Env.POSTGRES_SEEDS "" }}:{{ default .Env.DB_PORT "5432" }}"
+                connectProtocol: "tcp"
+                user: {{ default .Env.POSTGRES_USER "" }}
+                password: {{ default .Env.POSTGRES_PWD "" }}
+                maxConns: {{ default .Env.SQL_VIS_MAX_CONNS "20" }}
+                maxIdleConns: {{ default .Env.SQL_VIS_MAX_IDLE_CONNS "20" }}
+                maxConnLifetime: {{ default .Env.SQL_VIS_MAX_CONN_TIME "1h" }}
+        {{- end }}
+        {{- if eq $es "true" }}
+        es-visibility:
+            elasticsearch:
+                url:
+                    scheme: {{ default .Env.ES_SCHEME "http" }}
+                    host: "{{ default .Env.ES_SEEDS "" }}:{{ default .Env.ES_PORT "9200" }}"
+                username: {{ default .Env.ES_USER "" }}
+                password: {{ default .Env.ES_PWD "" }}
+                indices:
+                    visibility: {{ default .Env.ES_VIS_INDEX "temporal-visibility-dev" }} 
+        {{- end }}
+
+global:
+    membership:
+        maxJoinDuration: 30s
+        broadcastAddress: {{ default .Env.TEMPORAL_BROADCAST_ADDRESS "" }}
+    tls:
+        internode:
+            server:
+                requireClientAuth: true
+                certFile: /etc/temporal/config/certs/cluster/internode/cluster-internode.pem
+                keyFile: /etc/temporal/config/certs/cluster/internode/cluster-internode.key
+                requireClientAuth: true
+                clientCaFiles:
+                    - /etc/temporal/config/certs/cluster/ca/server-intermediate-ca-chain.pem
+            client:
+                serverName: internode.cluster-x.contoso.com
+                rootCaFiles:
+                    - /etc/temporal/config/certs/cluster/ca/server-intermediate-ca-chain.pem
+        frontend:
+            server:
+                requireClientAuth: true
+                certFile: /etc/temporal/config/certs/cluster/internode/cluster-internode.pem
```

Comment:
Ideally, we would create a 3rd certificate here and have the "cluster-level" public-facing certificate differ from the inter-node certificate, right? Otherwise it is not a good idea to have tctl using the cluster-internode certificate for security purposes

Reply:
We need to add a separate endpoint (port) for internal communication with frontend from within the cluster, so that it wouldn't have to be exposed to external network.